File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {

    const map = new Map(); 
    
    for(let name of completion) {
        map.set(name,(map.get(name)||0)+1);
    }
    
    for(let name of participant) {
        // map에 없는 이름이거나 ===0인 이름이면
        if(!map.has(name) || map.get(name) === 0) {
            return name;
        }
        map.set(name,map.get(name)-1);
    }
  
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        
        HashMap<String,Integer> map = new HashMap();
        for(String str: participant){
            map.put(str,map.getOrDefault(str,0)+1);
        }
        
        for(String str: completion){
            map.put(str, map.getOrDefault(str,0)-1);
        }
       
//         for(Map.Entry<String,Integer> entry: map.entrySet()){
//             String key = entry.getKey();
//             int value = entry.getValue();
//             System.out.println(key + " " +value);
            
//         }
        
        for(Map.Entry<String, Integer> entry: map.entrySet()){
            if(entry.getValue() == 1) {
                answer = entry.getKey();
            }
        }
      return answer;
    }
}